pennylane/decomposition/decomposition_graph.py

```diff
@@ -427,13 +456,36 @@ def __init__(self, graph: rx.PyDiGraph, original_op_indices: set[int], lazy: boo
         self.predecessors: dict[int, int] = {}
         self.unsolved_op_indices = original_op_indices.copy()
         self._num_edges_examined: dict[int, int] = {}  # keys are decomposition node indices
+        self._gate_set = gate_set
+
+    def _get_node_weight(self, gates):
+        """Calculates the weight of a node."""
+        node_weight = 0
+        if self._gate_set is not None:
+            for gate, count in gates.items():
+                gate_weight = 1.0
+                if gate in self._gate_set.keys():
+                    gate_weight = self._gate_set[gate]
+                node_weight += gate_weight * count
+        else:
+            for gate, count in gates.items():
+                node_weight += count
```

Comment:
I don't think `self._gate_set` is ever going to be `None`.
```suggestion
        node_weight = 0
        return sum(self._gate_set[gate] * count for gate, count in gates.items())      
```